### PR TITLE
Show 'pending' firmware with fixable problems when using 'fwupdmgr up…

### PR DESCRIPTION
**Pull request by hughsie**
…date'

Fixes https://github.com/fwupd/fwupd/issues/6853

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
Screenshot from 2024-02-28 12-15-32

**Comment by hughsie** [comment]
@mcatanzaro would something like the screenshot above have helped?

**Comment by hughsie** [comment]
> Does this change anything for GNOME Software or Plasma Discover?

Na. We cleverly (read: _accidentally_) stop behavior changes as we ask the client "what are you able to handle" -- so if a client sets FWUPD_FEATURE_FLAG_SHOW_PROBLEMS then we return results from the dbus methods like GetUpdates that can't actually be deployed at the moment, on the logic that older clients that don't know about checking for problems won't have that set.

But yes, as @superm1 says we need to fix that in those projects.

**Comment by mcatanzaro** [comment]
I've reported [this issue](https://gitlab.gnome.org/GNOME/gnome-software/-/issues/2465) for GNOME Software.

@Conan-Kudo please consider reporting a Plasma Discover bug.

> would something like the screenshot above have helped?

If it works without having to add the --force flag to show the unavailable updates, then yes.

It's fine to refuse to install available updates. Just not good to refuse to show them.

File: src/fu-tool.c [hunk 1]
```diff
@@ -1610,6 +1603,48 @@ fu_util_update(FuUtilPrivate *priv, gchar **values, GError **error)
...
+		for (guint i = 0; i < devices_unsupported->len; i++) {
+			FwupdDevice *dev = g_ptr_array_index(devices_unsupported, i);
+			fu_console_print(priv->console, " • %s", fwupd_device_get_name(dev));
+		}
+	}
+	if (devices_pending->len > 0) {
+		fu_console_print_literal(priv->console,
+					 /* TRANSLATORS: message letting the user there is an update
+					  * waiting, but there is a reason it cannot be deployed */
+					 _("Devices with pending firmware updates: "));
```
Comment (superm1) [inline src/fu-tool.c:1631]: Pending I feel like is the wrong adjective here.

Shouldn't it be more explicit?
Something like "devices with firmware updates available that can't be installed"?

**Review by superm1 — DISMISSED** [review]
Just one nit on the string to use for this, otherwise okay.

File: src/fu-tool.c (hunk 1 above)
Comment (hughsie) [reply src/fu-tool.c:1631]: That's a bit wordy. How about `Devices with updates that need user action:` or somethign like that

**Comment by Conan-Kudo** [comment]
> I've reported [this issue](https://gitlab.gnome.org/GNOME/gnome-software/-/issues/2465) for GNOME Software.
> 
> @Conan-Kudo please consider reporting a Plasma Discover bug.
> 

[This is done now](https://bugs.kde.org/show_bug.cgi?id=481943). FYI @aleixpol

File: src/fu-tool.c (hunk 1 above)
Comment (hughsie) [reply src/fu-tool.c]: Done!
